Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.h

```diff
@@ -36,126 +36,393 @@
+/*!
+ @abstract Checks the specified URL against the currently denied, granted, and requested permission match patterns in a specific tab.
+ @discussion URLs and match patterns can be granted on a per-tab basis. When the tab is known, access checks should always use this method.
+ @seealso permissionStateForURL:
+ @seealso hasAccessToURL:inTab:
+*/
+- (_WKWebExtensionContextPermissionState)permissionStateForURL:(NSURL *)url inTab:(nullable id <_WKWebExtensionTab>)tab NS_SWIFT_NAME(permissionState(forURL:in:));
 
+/*!
+ @abstract Sets the state of a URL in all tabs and global contexts with a distant furtue expiration date.
+ @discussion The URL is converted into a match pattern and will update `grantedPermissionMatchPatterns` and `deniedPermissionMatchPatterns`. Use this method for changing a single URL's state.
+ Only `WKWebExtensionContextPermissionStateDeniedExplicitly`, `WKWebExtensionContextPermissionStateUnknown`, and `WKWebExtensionContextPermissionStateGrantedExplicitly`
+ states are allowed to be set using this method.
+ @seealso setPermissionState:forURL:expirationDate:
+ @seealso setPermissionState:forURL:inTab:
+*/
+- (void)setPermissionState:(_WKWebExtensionContextPermissionState)state forURL:(NSURL *)url NS_SWIFT_NAME(setPermissionState(_:forURL:));
+
+/*!
+ @abstract Sets the state of a URL in all tabs and global contexts with a specific expiration date.
+ @discussion The URL is converted into a match pattern and will update `grantedPermissionMatchPatterns` and `deniedPermissionMatchPatterns`. Use this method for changing a single URL's state.
+ Passing a `nil` expiration date will be treated as a distant furtue date. Only `WKWebExtensionContextPermissionStateDeniedExplicitly`, `WKWebExtensionContextPermissionStateUnknown`,
+ and `WKWebExtensionContextPermissionStateGrantedExplicitly` states are allowed to be set using this method.
+ @seealso setPermissionState:forURL:
+ @seealso setPermissionState:forURL:inTab:
+*/
+- (void)setPermissionState:(_WKWebExtensionContextPermissionState)state forURL:(NSURL *)url expirationDate:(nullable NSDate *)expirationDate NS_SWIFT_NAME(setPermissionState(_:forURL:expirationDate:));
+
+/*!
+ @abstract Checks the specified match pattern against the currently denied, granted, and requested permission match patterns.
+ @discussion Match patterns can be granted on a per-tab basis. When the tab is known, access checks should always use the method that checks in a tab.
+ @seealso permissionStateForMatchPattern:inTab:
+ @seealso hasAccessToURL:inTab:
+*/
 - (_WKWebExtensionContextPermissionState)permissionStateForMatchPattern:(_WKWebExtensionMatchPattern *)pattern;
+
+/*!
+ @abstract Checks the specified match pattern against the currently denied, granted, and requested permission match patterns.
+ @discussion Match patterns can be granted on a per-tab basis. When the tab is known, access checks should always use the method that checks in a tab.
+ @seealso permissionStateForMatchPattern:inTab:
```

Comment:
I think there was a copy-past-o here. This is `permissionStateForMatchPattern:inTab:`. This comment should probably mention this is the method that checks the tab.